src/core/webview/ClineProvider.ts

```diff
@@ -2459,25 +2243,29 @@ export class ClineProvider implements vscode.WebviewViewProvider {
 	// global
 
 	async updateGlobalState(key: GlobalStateKey, value: any) {
-		await this.context.globalState.update(key, value)
+		this.outputChannel.appendLine(`Updating global state: ${key}`)
+		await this.contextProxy.updateGlobalState(key, value)
+
+		// // If we have a lot of pending changes, consider saving them periodically
+		// if (this.contextProxy.hasPendingChanges() && Math.random() < 0.1) { // 10% chance to save changes
+		// 	this.outputChannel.appendLine("Periodically flushing context state changes")
+		// 	await this.contextProxy.saveChanges()
+		// }
```

Comment:
Can you help me think through the tradeoffs here? What's the risk if we don't save? Is there a downside to saving every time like we used to, as long as the different windows read from the cache?

Reply:
@mrubens we still save on dispose (close vscode / or webview turn off), but if we turn on auto save with "Math.random() < 0.1" this will be have 10% chance to force it save

Reply:
What would the upside/downside be of saving on every change?

Reply:
Also trying to understand what the argument for saving a random 10% of the time would be

Reply:
> What would the upside/downside be of saving on every change?

@mrubens then we cannot using multiple instance of vscode, the best way is save it when the last instance close

Reply:
> Also trying to understand what the argument for saving a random 10% of the time would be

i think i will remove that line, that line is what Roo suggest me

Reply:
Sounds good!